pydecred/rpc.py

```diff
@@ -259,16 +494,322 @@ def parse(obj):
         )
 
 
+class SearchRawTransactionsResult:
+    """searchrawtransactions"""
+
+    def __init__(
+        self,
+        txid,
+        version,
+        lockTime,
+        expiry,
+        vin,
+        vout,
+        Hex=None,
+        blockHash=None,
+        blockHeight=None,
+        blockIndex=None,
+        confirmations=None,
+        time=None,
+        blockTime=None,
+    ):
+        """
+        Args:
+            txid (str):  The hash of the transaction.
+            version (int): The transaction version.
+            locktime (int): The transaction lock time.
+            expiry (int): The transacion expiry.
+            vin (list(object)): The transaction inputs as JSON objects.
+            vout (list(object)): The transaction outputs as JSON objects.
+            hex (str): Hex-encoded transaction or None.
+            blockHash (str): The hash of the block the contains the transaction or None.
+            blockHeight (int): The height of the block that contains the transaction or None.
+            blockIndex (int): The index within the array of transactions
+                contained by the block or None.
+            confirmations (int): Number of confirmations of the block or None.
+            time (int): Transaction time in seconds since 1 Jan 1970 GMT or None.
+            blockTime (int): Block time in seconds since the 1 Jan 1970 GMT or None.
+        """
+        self.txid = txid
+        self.version = version
+        self.lockTime = lockTime
+        self.expiry = expiry
+        self.vin = vin
+        self.vout = vout
+        self.hex = Hex
+        self.blockHash = blockHash
+        self.blockHeight = blockHeight
+        self.blockIndex = blockIndex
+        self.confirmations = confirmations
+        self.time = time
+        self.blockTime = blockTime
+
+    @staticmethod
+    def parse(obj):
+        return SearchRawTransactionsResult(
+            txid=obj["txid"],
+            version=obj["version"],
+            lockTime=obj["locktime"],
+            expiry=obj["expiry"],
+            vin=[Vin.parse(vin) for vin in obj["vin"]],
+            vout=[Vout.parse(vout) for vout in obj["vout"]],
+            Hex=get("hex", obj),
+            blockHash=get("blockhash", obj),
+            blockHeight=get("blockheight", obj),
+            blockIndex=get("blockindex", obj),
+            confirmations=get("confirmations", obj),
+            time=get("time", obj),
+            blockTime=get("blocktime", obj),
+        )
+
+
+class PrevOut:
+    """
+    PrevOut represents previous output for an input Vin.
+    """
+
+    def __init__(
+        self, value, addresses=[],
```

Comment:
Since lists are mutable, passing a list as a default value can cause some unintuitive behavior. https://repl.it/repls/HungryInnocentTypes

Instead, you can make the default value `None`, and then 

`self.addresses = addresses if addresses else []`